src/test/privileges.cpp

```diff
@@ -0,0 +1,193 @@
+
+using t_rawcaps = map<string,long int>; // raw caps as read from e.g. /proc/self/status, string nme => value (as integer)
+
+t_rawcaps read_slash_proc_rights() {
+	t_rawcaps caps;
+	std::string thefilename("/proc/self/status");
+	std::ifstream thefile(thefilename);
+	while (thefile.good()) {
+		string line="";
+		std::getline(thefile,line);
+		_dbg3("line: "<<line);
+		if (line.find("Cap")==0) {
+			_dbg2("has cap");
+			string sep=":\t";
+			size_t space_pos=line.find(sep);
+			if (space_pos != string::npos) {
+				string name(line.substr(0,space_pos));
+				bool name_ok=false;
+				if (name=="CapEff") name_ok=true;
+				if (name=="CapPrm") name_ok=true;
+				if (name=="CapInh") name_ok=true;
+				if (name=="CapBnd") name_ok=true;
+				if (name_ok) {
+					string data = line.substr(space_pos + sep.size());
+					t_rawcaps::mapped_type data_int;
+					std::istringstream iss(data);
+					iss >> std::hex >> data_int;
+					bool ok = ( (!iss.fail()) && (iss.eof()) );
+					if (!ok) _throw_error_runtime("Can not parse Caps (from thefile "s + thefilename + ") - the line is: ["s + line + "]"s );
+					caps[name] = data_int;
+				} // name
+			} // sep
+		} // Cap
+	} // loop all lines of status
+	_dbg1("Read caps from "<<thefilename<<" - size: " << caps.size());
+	return caps;
+}
+
+} // namespace detail
+
+
+TEST( privileges , name_cap_numbers ) {
+	EXPECT_EQ( static_cast<long int>(capmodpp::get_last_cap_nr()) , static_cast<long int>(CAP_LAST_CAP) );
+	EXPECT_EQ( static_cast<long int>(capmodpp::get_cap_size()) , static_cast<long int>(CAP_LAST_CAP + 1) );
+	EXPECT_GT( static_cast<long int>(capmodpp::get_cap_size()) , 5); // there are always more then this CAPs
+}
+
+TEST( privileges , name_and_capnr_conversions ) {
+	// example cap number that does not exist
+	// need to update this example if kernel would include such cap in future
+	EXPECT_THROW( { auto x = capmodpp::cap_nr_to_name(100); volatile auto dump = x; } , std::runtime_error );
+
+	EXPECT_THROW( { auto x = capmodpp::cap_nr_to_name(CAP_LAST_CAP + 1 ); volatile auto dump = x; } , std::runtime_error );
+	EXPECT_THROW( { auto x = capmodpp::cap_nr_to_name( capmodpp::get_last_cap_nr() + 1 ); volatile auto dump = x; } , std::runtime_error );
+	EXPECT_NO_THROW( { auto x = capmodpp::cap_nr_to_name(CAP_LAST_CAP); volatile auto dump = x; } );
+	EXPECT_NO_THROW( { auto x = capmodpp::cap_nr_to_name( capmodpp::get_last_cap_nr()); volatile auto dump = x; } );
+
+	// example cap name that does not exist
+	// need to update this example if kernel would include such cap in future :)
+	EXPECT_THROW( capmodpp::cap_name_to_nr("MINE_BITCOINS") , std::runtime_error );
+	EXPECT_NO_THROW( capmodpp::cap_name_to_nr("SYS_ADMIN") );
+}
+
+
+TEST( privileges , create_print_capmodpp ) {
+	capmodpp::cap_state_map state_map1;
+	_note("State map empty: " << state_map1 );
+
+	auto slashproc1 = detail::read_slash_proc_rights();
+	for (const auto & item : slashproc1) _note(item.first << " = " << std::hex << item.second);
```

Comment:
:no_entry: this debugs are helpfull

Note: integration-like tests, will be later moved to other test binary for clarity